pkg/reconcile/common.go

```diff
@@ -140,16 +141,37 @@ func (u *MonitorResourceCommon) UpdateMonitorResourceStatus(cr client.Object) (r
 	return reconcile.StopReconcile()
 }
 
-func (u *MonitorResourceCommon) GetClusterID() string {
-	if u.ClusterID == "" {
-		var version configv1.ClusterVersion
-		err := u.Client.Get(u.Ctx, client.ObjectKey{Name: "version"}, &version)
-		if err != nil {
-			return ""
-		}
-		u.ClusterID = string(version.Spec.ClusterID)
+// GetOSDClusterID returns the ID for the cluster based on its ClusterVersion
+func (u *MonitorResourceCommon) GetOSDClusterID() (string, error) {
+	var version configv1.ClusterVersion
+	err := u.Client.Get(u.Ctx, client.ObjectKey{Name: "version"}, &version)
+	if err != nil {
+		return "", err
 	}
-	return u.ClusterID
+	return string(version.Spec.ClusterID), nil
+}
+
+// GetHypershiftClusterID returns the ID for a hosted cluster based on the HCP object in the same namespace as the provided ClusterURLMonitor
+func (u *MonitorResourceCommon) GetHypershiftClusterID(ns string) (string, error) {
+	hcp, err := u.GetHCP(ns)
+	if err != nil {
+		return "", err
+	}
+	return hcp.Spec.ClusterID, nil
+}
+
+// GetHCP returns the HostedControlPlane object in the namespace provided. If more than one HCP object exists in the same namespace, an error is returned
+func (u *MonitorResourceCommon) GetHCP(ns string) (hypershiftv1beta1.HostedControlPlane, error) {
+	// Retrieve the HostedControlPlane in order to lookup the associated hostedCluster object
```

Comment:
Can implement backoff retries here in the future as HCP may not exist for some time

Reply:
To be clear, are you suggesting this be added now, or just making a note to add this in the future?

Reply:
Just making note to add this in the next iteration